core/image.cpp

```diff
@@ -1915,8 +1917,10 @@ void Image::create(int p_width, int p_height, bool p_use_mipmaps, Format p_forma
 }
 
 void Image::create(int p_width, int p_height, bool p_use_mipmaps, Format p_format, const Vector<uint8_t> &p_data) {
-	ERR_FAIL_INDEX(p_width - 1, MAX_WIDTH);
-	ERR_FAIL_INDEX(p_height - 1, MAX_HEIGHT);
+	ERR_FAIL_COND_MSG(p_width <= 0, "Image width must be greater than 0.");
+	ERR_FAIL_COND_MSG(p_height <= 0, "Image height must be greater than 0.");
+	ERR_FAIL_COND_MSG(p_width > MAX_WIDTH, "Image width cannot be greater than " + itos(MAX_WIDTH) + ".");
+	ERR_FAIL_COND_MSG(p_height > MAX_HEIGHT, "Image height cannot be greater than " + itos(MAX_HEIGHT) + ".");
 	ERR_FAIL_COND_MSG(p_width * p_height > MAX_PIXELS, "Too many pixels for image, maximum is " + itos(MAX_PIXELS));
```

Comment:
I wonder how `MAX_WIDTH` and `MAX_HEIGHT` relate to `MAX_PIXELS`. Shouldn't it be possible to create an image which exceed the maximum width/height as long as the total number of pixels does not exceed `MAX_PIXELS` (for instance 20000x1024)? I know that the rationale behind max width/height is mandated by GPU texture size restrictions, but image processing can have other purposes other than simply feeding an image to GPU.

Reply:
`MAX_PIXELS` seems to be `16384 * 16384`, and `16384` is a typical value given by GPUs for `GL_MAX_TEXTURE_SIZE`.

Reply:
This was changed in da0457fa29e, and `MAX_PIXELS` used to be `MAX_WIDTH * MAX_HEIGHT`, but evidently @reduz needed more.

Reply:
Ah I've just checked those constants, both have `1 << 24` (16 777 216‬), basically unrestricted in image width and height, as long as it doesn't exceed `MAX_PIXELS`, I'm satisfied in fact. 😛

Reply:
> Ah I've just checked those constants, both have `1 << 24` (16 777 216‬), basically unrestricted in image width and height, as long as it doesn't exceed `MAX_PIXELS`, I'm satisfied in fact. 😛

Are this only changed in 4.x, not 3.x?